extensions/ql-vscode/src/local-queries/local-query-run.ts

```diff
@@ -61,6 +62,10 @@ export class LocalQueryRun {
     private readonly localQueries: LocalQueries,
     private readonly queryInfo: LocalQueryInfo,
     private readonly dbItem: DatabaseItem,
+    /**
+     * The logger is only available while the query is running and will be disposed of when the
+     * query completes.
+     */
     public readonly logger: Logger, // Public so that other clients, like the debug adapter, know where to send log output
```

Comment:
I believe that we always pass a logger that is disposable. Is it worth putting that in the type here instead of using `isDisposable` in every place?

If we don't want to define a new `DispoableLogger` type we could instead do `logger: Logger & Disposable` here.

Reply:
Good idea, I've used `Logger & Disposable` here and removed the `isDisposable` function.